lib/internal/util/inspect.js

```diff
@@ -2000,6 +2013,88 @@ function stripVTControlCharacters(str) {
   return str.replace(ansi, '');
 }
 
+function getPrivatePropertyPreviews(ctx, target) {
+  const preview = previewValue(target);
+  if (preview == null) {
+    return undefined;
+  }
+  const { privateProperties } = preview;
+  if (privateProperties == null) {
+    return undefined;
+  }
+  return privateProperties.map((it) => formatPrivatePropertyPreview(ctx, it));
+}
+
+function previewPrelude() {
+  if (!hasInspector) {
+    return false;
+  }
+  if (privateInspectionSession == null) {
+    inspector = require('internal/inspector');
+    privateInspectionSession = new inspector.SyncSession();
+    privateInspectionSession.connect();
+  }
+  return true;
+}
+
+function previewValue(target) {
+  if (!previewPrelude()) {
+    return undefined;
+  }
+  privateInspectionSession.post('Runtime.enable');
+
+  /** Attach the target to globalThis to fetch its objectId. */
+  globalThis.__util_inspection_target__ = target;
+  const { result: evalResult } = privateInspectionSession.post(
+    'Runtime.evaluate', {
+      expression: 'globalThis.__util_inspection_target__',
+    });
+  delete globalThis.__util_inspection_target__;
+  /** Side effects cleaned. */
+
+  if (evalResult.type !== 'object') {
+    return;
+  }
+  const previews = privateInspectionSession.post('Runtime.getProperties', {
+    objectId: evalResult.objectId,
+  });
+  privateInspectionSession.post('Runtime.disable');
+  return previews;
+}
+
+function formatPrivatePropertyPreview(ctx, preview) {
+  const valueDescriptor = preview.value;
+  const valueType = valueDescriptor.type;
+  const stylize = ctx.stylize || stylizeNoColor;
+  let str;
+  if (valueType === 'string') {
+    str = stylize(`'${valueDescriptor.value}'`, valueType);
+  } else if (valueType === 'number') {
+    str = stylize(valueDescriptor.value, valueType);
+  } else if (valueType === 'boolean') {
+    str = stylize(valueDescriptor.value, valueType);
+  } else if (valueType === 'symbol') {
+    str = stylize(valueDescriptor.description, valueType);
+  } else if (valueType === 'undefined') {
+    str = stylize(undefined, valueType);
```

Comment:
We have to pass through strings to the stylize function.

```suggestion
    str = stylize('undefined', valueType);
```